crates/node/metrics/src/chain.rs

```diff
@@ -0,0 +1,19 @@
+//! This exposes reth's chain information over prometheus.
+use metrics::{describe_gauge, gauge};
+
+/// Contains chain information for the application.
+#[derive(Debug, Clone)]
+pub struct ChainSpecInfo {
```

Comment:
this should use `reth-metrics-derive`

Reply:
Looked into this one and I don't think it's possible. The thing is that this is a "fake" metric where we create a `gauge` that is only used to set labels on it. The `gauge` itself is never used. It doesn't look like `reth-metrics-derive` supports this case.